[thienmy0] iP

thienmy0 opened the pull request:
# Bob Bot :robot:
> A wise man once said "Sleep is for the weak."
- Not the **fastest**
- Not the _most efficient_
- ~~But has the spirit~~

Some things to improve on:
1. Parser
2. OOP
3. Essentially Everything

```ruby
Use it to keep track of your tasks if you want to
```

- [x] Manage Todos
- [ ] Manage Deadlines
- [ ] Manage Events

[bop](https://youtu.be/LTRWgZjRViw?si=j6kSSd-BPSG-17Ll)

This is an `inline code` I think

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

thienmy0 committed:
Level 0

thienmy0 committed:
Level 1

thienmy0 committed:
Level 2

thienmy0 committed:
Level 3

thienmy0 committed:
Level 4

thienmy0 committed:
Text Ui Testing

thienmy0 committed:
Level 5

thienmy0 committed:
Level 6

thienmy0 committed:
Level 7. Save

thienmy0 committed:
Merge pull request #1 from thienmy0/branch-Level-7

Level 7. Save

thienmy0 committed:
Level 8. Dates and Times

thienmy0 committed:
Merge pull request #2 from thienmy0/branch-Level-8

Level 8. Dates and Times

thienmy0 committed:
MoreOOP

thienmy0 committed:
A-Packages

thienmy0 committed:
Merge pull request #3 from thienmy0/add-gradle-support

Add gradle support

thienmy0 committed:
Set up Gradle

thienmy0 committed:
JUnit Tests

thienmy0 committed:
Javadoc

thienmy0 committed:
Coding Standard

thienmy0 committed:
Level 9. Find

thienmy0 committed:
Merge pull request #4 from thienmy0/branch-A-JavaDoc

Javadoc

thienmy0 committed:
Merge branch 'master' into branch-A-CodingStandard

thienmy0 committed:
Merge pull request #5 from thienmy0/branch-A-CodingStandard

Coding Standard

thienmy0 committed:
Merge pull request #6 from thienmy0/branch-Level-9

Level 9. Find

thienmy0 committed:
Update Ui.java

aslam341 commented on src/main/java/bob/Storage.java:
```diff
@@ -0,0 +1,142 @@
+     * @return ArrayList</Task> the list of tasks that were recorded in the existing file, else an empty list
+     * @throws BobException
+     */
+    public ArrayList<Task> load() throws BobException {
```
Should some parts in this method be extracted out? As per [code quality guidelines](https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html), we should avoid long methods.

aslam341 commented on src/main/java/bob/Ui.java:
```diff
@@ -0,0 +1,208 @@
+     * @param task the input string given
+     * @throws BobException
+     */
+    public static void checkAndAddTask(TaskList list, String task) throws BobException {
```
Should some parts in this method be extracted out? As per [code quality guidelines](https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html), we should avoid long methods.

aslam341 commented on src/main/java/bob/Parser.java:
```diff
@@ -0,0 +1,99 @@
+     * @param input
+     * @return boolean true if input is find command
+     */
+    public boolean isFind(String input) {
```
Well done on naming boolean methods to sound like boolean here, and in the rest of the places.

aslam341 commented on src/main/java/bob/Parser.java:
```diff
@@ -0,0 +1,99 @@
+     */
+    public boolean isDelete(String input) {
+        char[] charArray = input.toCharArray();
+        if (charArray[0] == 'd' && charArray[1] == 'e' && charArray[2] == 'l' && charArray[3] == 'e' &&
```
Perhaps there is a better way to do this than brute-force it here and in other places? You could look into regex (regular expression) or other methods of doing this easier, so as to avoid complicated expressions which is one of the guidelines for [code quality](https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html).

aslam341 commented on src/main/java/bob/Storage.java:
```diff
@@ -0,0 +1,142 @@
+public class Storage {
+
+    String filePath;
+    //Parser parser;
```
Should this be deleted? This is considered dead code.

thienmy0 committed:
Checkstyle

thienmy0 committed:
Merge pull request #7 from thienmy0/branch-A-CheckStyle

Checkstyle

thienmy0 committed:
GUI Addition 1

thienmy0 committed:
Fix GUI Implementations & Improvements

thienmy0 committed:
Merge pull request #8 from thienmy0/branch-Level-10

Level 10

Clin-lyx commented on src/main/java/bob/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDate by;
```
Perhaps a more meaningful name would be better?

Clin-lyx commented on src/main/java/bob/Parser.java:
```diff
@@ -0,0 +1,96 @@
+     */
+    public boolean isMark(String input) {
+        char[] charArray = input.toCharArray();
+        if (charArray[0] == 'm' && charArray[1] == 'a' && charArray[2] == 'r' && charArray[3] == 'k'
```
I think there is a startsWith method that would do the same functionality (e.g. input.startsWith("mark")).

Clin-lyx commented on src/main/java/bob/Storage.java:
```diff
@@ -0,0 +1,146 @@
+
+        //creates file if file doesn't exist, else reads file and re-initialises tasks
+        try {
+            if (file.createNewFile()) {
+                //file is created
+                System.out.println("Hello! I'm Bob");
+                System.out.println("What can I do for you?");
+            } else {
+                //file exists
+                //read the file and re-initiate the list of tasks
+                Scanner s = new Scanner(file);
+                int index = 0;
```
I think this part can be extracted out as a createFile method :)

Clin-lyx commented on src/main/java/bob/Storage.java:
```diff
@@ -0,0 +1,146 @@
+                    int firstDate = charArray.length;
+                    int secondDate = charArray.length;
+
+                    for (int i = 4; i < charArray.length; i++) {
+                        if (charArray[i] != ',' && i < firstDate) {
+                            taskName += charArray[i];
+                        }
+
+                        if (charArray[i] == ',' && firstDate == secondDate) {
+                            firstDate = i;
+                        } else if (charArray[i] == ',' && firstDate != secondDate) {
+                            secondDate = i;
+                        }
+
+                        if (i > firstDate && i < secondDate) {
+                            dateOne += charArray[i];
+                        } else if (i > secondDate) {
+                            dateTwo += charArray[i];
+                        }
+                    }
+
+                    if (charArray[0] == 'T') {
+                        list.add(new Todo(taskName));
+                    } else if (charArray[0] == 'D') {
+                        LocalDate d1 = LocalDate.parse(dateOne);
+                        list.add(new Deadline(taskName, d1));
+                    } else if (charArray[0] == 'E') {
+                        LocalDate d1 = LocalDate.parse(dateOne);
+                        LocalDate d2 = LocalDate.parse(dateTwo);
+                        list.add(new Event(taskName, d1, d2));
+                    }
+
+                    if (charArray[2] == '1') {
+                        list.get(index).markAsDone();
+                    }
+
+                    index++;
+                }
+                return list;
+            }
```
I think it would be better to extract out this part as a readFile method.

Clin-lyx commented on src/main/java/bob/Ui.java:
```diff
@@ -0,0 +1,243 @@
+     */
+    public static String checkAndAddTask(TaskList list, String task) throws BobException {
+        char[] charArray = task.toCharArray();
+        String taskName = "";
```
Maybe taskDescription would be a better variable name.

Clin-lyx reviewed (commented):
LGTM. I think all the code follows the coding standard properly. Might need some extra effort to make methods not so lengthy and readable.

thienmy0 committed:
Add assertions to code

Add assertions to code

This project contains many classes that work together.

These classes are prone to errors.

Let's add assertions to some parts of the code to:
- indicate possible bugs
- ensure correctness of code

If the runtime detects an assertion failure, it typically takes some
drastic action such as terminating the execution with an error message.

thienmy0 committed:
Merge pull request #9 from thienmy0/branch-A-Assertions

Add assertions to code

thienmy0 committed:
Improve Code Quality

Improve Code Quality

The code is messy and complicated.

Making the code more readable is crucial for other developers to
understand the code, and ensure that we will have greater clarity with
the code in the future.

Let's improve code quality by avoiding long methods, complicated
expressions and practice KISSing.

Doing so makes the code easier on the eyes to see and understand.

thienmy0 committed:
Merge pull request #10 from thienmy0/branch-A-CodeQuality

Improve Code Quality

thienmy0 committed:
Add Extension B-Snooze

thienmy0 committed:
Improve organisation and add more error handling

thienmy0 committed:
Add more personality

thienmy0 committed:
Improve GUI

thienmy0 committed:
Add more testing

thienmy0 committed:
Add some minor improvements

thienmy0 committed:
Add user guide and minor improvements

thienmy0 committed:
Add final improvements

thienmy0 committed:
Update Ui class

thienmy0 committed:
Update build.gradle

thienmy0 committed:
Update build.gradle
